src/nvim/window.c

```diff
@@ -5048,27 +5068,22 @@ int min_rows(void)
   return total;
 }
 
-/*
- * Return TRUE if there is only one window (in the current tab page), not
- * counting a help or preview window, unless it is the current window.
- * Does not count "aucmd_win".
- */
-int only_one_window(void)
+/// Return true if there is only one window (and only one tab page), not
+/// counting a help or preview window, unless it is the current window.
+/// Does not count "aucmd_win".
+bool only_one_window(void) FUNC_ATTR_WARN_UNUSED_RESULT
```

Comment:
I also think this is PURE. Note that pure attribute has actually nothing to do with the term “pure function” from computer science. Function marked like PURE should not have any side-effects, but it may access global state even if global state is not constant. Attribute is used for some optimizations like common subexpression elimination.

So it is not nonsense to have PURE function with `(void)` arguments which can return different results on different calls.